docs/generating-tests.rst

```diff
@@ -0,0 +1,417 @@
+to fill tests.  ``--checkstate`` tells ``testeth`` to check the final states against
+the ``expect`` fields.
+
+Depending on your shell, there are various ways to set up ``ETHEREUM_TEST_PATH``
+environment variable.  For example, adding ``export ETHEREUM_TEST_PATH=/path/to/tests``
+to ``~/.bashrc`` might work for ``bash`` users.
+
+``testeth`` with ``--filltests`` fills every test filler it finds. The command might
+modify existing test cases. After running ``testeth`` with ``--filltests``\ , try running
+``git status`` in the ``tests`` directory. If ``git status`` indicates changes in 
+unexpected files, that is an indication that the behavior of ``cpp-ethereum`` changed unexpectedly.
+
+.. note::
+   If ``testeth`` is looking for tests in the ``../../test/jsontests`` directory, 
+   you have probably not specified the ``--testpath`` option.
+
+
+Trying the Filled Test
+----------------------
+
+Trying the Filled Test Locally
+++++++++++++++++++++++++++++++
+
+For trying the filled test, in ``aleth/build`` directory, run
+
+.. code:: bash
+
+   ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2
+
+
+Trying the Filled Test in Travis CI
++++++++++++++++++++++++++++++++++++
+
+Moreover, for trying the filled test in ``Travis CI`` for ``ethereum/aleth``, 
+the new test cases need to exist in a branch in ``ethereum/tests``. For this,
+ask somebody with a push permission to ``ethereum/tests``.
+
+After that, enter ``aleth/test/jsontests`` directory, and checkout the branch in
+``ethereum/tests``.  Then go back to ``aleth`` directory and perform 
+``git add test/jsontests`` followed by ``git commit``.
+
+When you file this commit as a pull-request to ``ethereum/aleth``, Travis CI 
+should try the newly filled tests.
+
+git commit
+----------
+
+After these are successful, the filler file and the filled test should be added 
+to the ``tests`` repository. File these as a pullrequest.
+
+If changes in the cpp-client were necessary, also file a pull-request there.
+
+Advanced: Converting a GeneralStateTest Case into a BlockchainTest Case
+=======================================================================
+
+In the tests repository, each GeneralStateTest is eventually translated into a 
+BlockchainTest.  This can be done by the following sequence of commands.
+
+.. code::
+
+  ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2 -- --filltests --fillchain --checkstate
+
+
+followed by
+
+.. code::
+
+  ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2 -- --filltests --checkstate
+
+
+The second command is necessary because the first command modifies the GeneralStateTests in an undesired way.
+
+After these two commands,
+
+
+* ``git status`` to check if any GeneralStateTest has changed.  If yes, revert the changes, and follow section _\ ``Trying the Filled Test Locally``.  That will probably reveail an error that you need to debug.
```

Comment:
This will also be affected by the way the testeth and lllc version is included in the generated files. It works out ok for new tests though since they turn up as new files and the other changes are all to existing files.

Given the move to specific releases for the reference tests, I wonder if it makes sense to leave generating the blockchain versions of tests to a one-off run before taking the release.

Reply:
Will leave this as a suggestion here for now, maybe @winsvega wants to add on this as well.

Reply:
Will leave this for now. I think this is due to a larger rewrite which can be done in follow-up PRs.